flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/SlotAllocationTest.java

```diff
@@ -40,17 +48,142 @@ public void test() {
 			public boolean filter(Long value) { return false; }
 		};
 
-		env.generateSequence(1, 10).filter(dummyFilter).isolateResources().filter(dummyFilter)
-				.disableChaining().filter(dummyFilter).startNewResourceGroup().filter(dummyFilter)
-				.startNewChain().print();
+		env.generateSequence(1, 10)
+			.filter(dummyFilter).slotSharingGroup("isolated")
+			.filter(dummyFilter).slotSharingGroup("default").disableChaining()
+			.filter(dummyFilter).slotSharingGroup("group 1")
+			.filter(dummyFilter).startNewChain()
+			.print().disableChaining();
+
+		// verify that a second pipeline does not inherit the groups from the first pipeline
+		env.generateSequence(1, 10)
+				.filter(dummyFilter).slotSharingGroup("isolated-2")
+				.filter(dummyFilter).slotSharingGroup("default").disableChaining()
+				.filter(dummyFilter).slotSharingGroup("group 2")
+				.filter(dummyFilter).startNewChain()
+				.print().disableChaining();
 
 		JobGraph jobGraph = env.getStreamGraph().getJobGraph();
 
 		List<JobVertex> vertices = jobGraph.getVerticesSortedTopologicallyFromSources();
 
-		assertEquals(vertices.get(0).getSlotSharingGroup(), vertices.get(2).getSlotSharingGroup());
+		assertEquals(vertices.get(0).getSlotSharingGroup(), vertices.get(3).getSlotSharingGroup());
+		assertNotEquals(vertices.get(0).getSlotSharingGroup(), vertices.get(2).getSlotSharingGroup());
+		assertNotEquals(vertices.get(3).getSlotSharingGroup(), vertices.get(4).getSlotSharingGroup());
+		assertEquals(vertices.get(4).getSlotSharingGroup(), vertices.get(5).getSlotSharingGroup());
+		assertEquals(vertices.get(5).getSlotSharingGroup(), vertices.get(6).getSlotSharingGroup());
```

Comment:
That's how I found the ordering out. Apparently first the sources appear and then the rest of the individual pipelines. But here the first pipeline comes first. 

Given that a topological ordering is partial and, thus, there are multiple linearization of it, I think this test is bound to crash at some point. It is simply too tightly coupled to the internal implementation of the `JobGraph`.